test/scripts/kind_dual_conversion.sh

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+set -x
+
+convert_kubelet() {
+ echo "Converting kubelet on nodes $NODES"
+ for n in $NODES; do
+  # /var/lib/kubelet/config.yaml
+  # cat <<EOF >> "/var/lib/kubelet/config.yaml"
+# featureGates:
+#   IPv6DualStack: true
+# EOF
+   docker exec $n cp /var/lib/kubelet/config.yaml /var/lib/kubelet/config.yaml.bak
+   docker exec $n sed -i '/featureGates/a\  IPv6DualStack: true' /var/lib/kubelet/config.yaml
+   # Restart kubelet
+   docker exec $n systemctl restart kubelet
+  echo "Converted kubelet on node $n"
+ done
+}
+
+convert_cni() {
+  # from kind hack/ci/e2e-k8s.sh
+  # Modify current OVN config
+  original_ovnconfig=$(kubectl get -oyaml -n=ovn-kubernetes configmap/ovn-config)
+  echo "Original OVN config:"
+  echo "${original_ovnconfig}"
+  # Patch it
+  fixed_ovnconfig=$(
+    printf '%s' "${original_ovnconfig}" | sed \
+      -e "s#^  net_cidr.*#&\,${SECONDARY_CLUSTER_SUBNET}#" \
+      -e "s#^  svc_cidr:.*#&\,${SECONDARY_SERVICE_SUBNET}#" \
+  )
+  echo "Patched OVN config:"
+  echo "${fixed_ovnconfig}"
+  printf '%s' "${fixed_ovnconfig}" | kubectl apply -f -
+  # restart ovnkube-master
+  kubectl -n ovn-kubernetes rollout restart deployment ovnkube-master
+  kubectl -n ovn-kubernetes rollout status deployment ovnkube-master
+  # restart ovnkube-node
+  kubectl -n ovn-kubernetes rollout restart daemonset ovnkube-node
+  kubectl -n ovn-kubernetes rollout status daemonset ovnkube-node
+  echo "Updated CNI"
+}
+
+convert_control_plane(){
+  # Kubeadm installs apiserver and controller-manager as static pods
+  # one the configuration has changed kubelet restart them
+  API_CONFIG_FILE="/etc/kubernetes/manifests/kube-apiserver.yaml"
+  CM_CONFIG_FILE="/etc/kubernetes/manifests/kube-controller-manager.yaml"
+
+  # update all the control plane nodes
+  for n in $CONTROL_PLANE_NODES; do
+    echo "Converting control-plane on node $n"
+    # kube-apiserver backup config file
+    docker exec $n cp ${API_CONFIG_FILE} /kind/${API_CONFIG_FILE}.bak
+    # append second service cidr --service-cluster-ip-range=<IPv4 CIDR>,<IPv6 CIDR>
+    docker exec $n sed -i -e "s#.*service-cluster-ip-range.*#&\,${SECONDARY_SERVICE_SUBNET}#" ${API_CONFIG_FILE}
+    # configure feature gate --feature-gates=IPv6DualStack=true
+    # docker exec $n sed -i '/service-cluster-ip-range/i\    - --feature-gates=IPv6DualStack=true' ${API_CONFIG_FILE}
```

Comment:
don't need this comment right?

Reply:
still WIP, had to try different sed approaches ... yaml in bash :fire: :face_with_head_bandage: